src/test/ui/pattern/bindings-after-at/borrowck-pat-by-copy-bindings-in-at.rs

```diff
@@ -0,0 +1,52 @@
+// check-pass
+
+// Test `Copy` bindings in the rhs of `@` patterns.
+
+#![feature(slice_patterns)]
+#![feature(bindings_after_at)]
+
+#[derive(Copy, Clone)]
+struct C;
+
+fn mk_c() -> C { C }
+
+#[derive(Copy, Clone)]
+struct P<A, B>(A, B);
+
+enum E<A, B> { L(A), R(B) }
+
+fn main() {
+    let a @ b @ c @ d = C;
+    let a @ (b, c) = (C, mk_c());
+    let a @ P(b, P(c, d)) = P(mk_c(), P(C, C));
+    let a @ [b, c] = [C, C];
+    let a @ [b, .., c] = [C, mk_c(), C];
+    let a @ [b, mid @ .., c] = [C, mk_c(), C];
+    let a @ &(b, c) = &(C, C);
+    let a @ &(b, &P(c, d)) = &(mk_c(), &P(C, C));
+
+    fn foo(a @ [b, mid @ .., c]: [C; 3]) {}
+
+    use self::E::*;
+    match L(C) {
+        L(a) | R(a) => {
+            let a: C = a;
+            drop(a);
+            drop(a);
```

Comment:
`drop(a); drop(a);` (for `A: C` where `C: Copy`) ? Were these tests auto-generated or something?

Reply:
Nothing auto-generated in this PR. :)

The `drop(a); drop(a);` is just to ensure that nothing is out of the ordinary wrt. a `Copy` type in this context.

Reply:
I don't understand the motivation (especially for writing it twice), but it does not bother me as written.